cpp/include/cudf/types.hpp

```diff
@@ -24,8 +24,12 @@
 #define CUDA_DEVICE_CALLABLE inline
 #endif
 
+#include <thrust/optional.h>  // TODO no idea why this is needed ¯\_(ツ)_/¯
```

Comment:
Still needed?

Reply:
Unfortunately yes. I will revisit and fix (have set up https://github.com/rapidsai/cudf/issues/6475)